src/Language/PureScript/Errors.hs

```diff
@@ -763,16 +764,24 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
       line $ "Type of expression contains additional label " <> markCode (prettyPrintLabel prop) <> "."
     renderSimpleErrorMessage TypeSynonymInstance =
       line "Type class instances for type synonyms are disallowed."
-    renderSimpleErrorMessage (OrphanInstance nm cnm ts) =
-      paras [ line $ "Type class instance " <> markCode (showIdent nm) <> " for "
+    renderSimpleErrorMessage (OrphanInstance nm cnm nonOrphanModules ts) =
+      paras [ line $ "Orphan instance " <> markCode (showIdent nm) <> " found for "
             , markCodeBox $ indent $ Box.hsep 1 Box.left
                 [ line (showQualified runProperName cnm)
                 , Box.vcat Box.left (map typeAtomAsBox ts)
                 ]
-            , line "is an orphan instance."
-            , line "An orphan instance is one which is defined in a module that is unrelated to either the class or the collection of data types that the instance is defined for."
-            , line "Consider moving the instance, if possible, or using a newtype wrapper."
-            ]
+            , Box.vcat Box.left $ case modulesToList of
+                [] -> [ line "There is nowhere this instance can be placed without being an orphan."
+                      , line "A newtype wrapper can be used to avoid this problem."
```

Comment:
Seems like we'd also probably need a link to what the hell a newtype wrapper is and how to define it?

Reply:
Spoke to @paf31, suggested that this level of detail would be better suited to the wiki. We're going to talk a bit more about this later, and I'll probably make another docs PR :)